src/animations.ts

```diff
@@ -111,105 +455,107 @@ let FAILED_KARAOKE_UPDATE_ATTEMPTS = 0
  * Even with the correct import order and with the `defer` attribute applied, `updateKaraoke` may not be ready to be
  * invoked by the time the animation starts.
  */
-function attemptUpdateKaraoke(karaokeAnimationInfo: KaraokeAnimationInfo, animationTime: number) {
+function attemptUpdateKaraoke(karaokeAnimationInfo: KaraokeAnimationInfo, animationTime: number, karaokeState: KaraokeState | null) {
     if (typeof updateKaraoke !== 'undefined') {
-        updateKaraoke(karaokeAnimationInfo, animationTime)
+        return updateKaraoke(karaokeAnimationInfo, animationTime, karaokeState)
     } else {
         FAILED_KARAOKE_UPDATE_ATTEMPTS++
         // Assuming the animation runs at 60 FPS, this will give about ten seconds of cushion before sending a Sentry
         // error.
         if (FAILED_KARAOKE_UPDATE_ATTEMPTS === 600) {
             Sentry.captureMessage('`updateKaraoke` was not loaded properly.', 'warning')
         }
+        return null
     }
 }
 
 
 function setRadialProgressBar(animation: AnimationInfo, animationTime: number) {
     const audioProgress = animationTime / animation.duration
-    const audioProgressBar = $<SVGCircleElement>(animation.progressBarSelector)
+    const audioProgressBar = $(animation.progressBar)
     const strokeOffset = (1 - audioProgress) * 2 * Math.PI * parseInt(audioProgressBar.attr('r'))
     audioProgressBar.css({ strokeDashoffset: strokeOffset })
 }
 
 
-function heroAnimation() {
-    const animation = ANIMATIONS[HERO_VIDEO_ANIMATION]
-
-    setRadialProgressBar(animation, HERO_VIDEO.currentTime * 1000)
+function sampleQuestionAnimation(animationName: AnimationName, karaokeState: KaraokeState = null) {
+    const animation = ANIMATIONS[animationName]
+    const isSameAudio = (PLAYER.querySelector('source').src === animation.expectedAudioSrc)
+    const isSameAudioPlaying = isSameAudio && !PLAYER.paused
```

Comment:
Just making sure playing and !paused are equivalent? (in case there are other statuses like stopped, etc.)

Reply:
Just double checked to be sure, there isn't a `stopped` status or anything like that. And `paused` will be true before playback ever begins, if the audio is paused midway through, and after playback is completed 🙂